### ISSUE TITLE
Home mini electricity data no longer updating every minute

### ISSUE THREAD

**Issue by paulmthomas2** [BottlecapDave/HomeAssistant-OctopusEnergy#914]
### Describe the bug

sensor.octopus_energy_electricity_REDACT_current_demand (linked to the Home mini) is no longer updating each minute. The API would generally seem to be working as Octopus App and octoaid are showing live data, but this sensor seems only to be picking up new values c. every 30 minutes

### Reproduction steps

ongoing

### Expected behaviour

expect updates every minute (sensor is currently calling every minute still, but not returning new data)

### Tariff Code

 E-1R-INTELLI-VAR-22-10-14-H

### Integration Version

11.1.1

### Home Assistant Version

2024.6.2

### Fresh Install?

Not specified

### Home Assistant Logs

[home-assistant_octopus_energy_2024-06-14T15-40-18.911Z.log](https://github.com/user-attachments/files/15840538/home-assistant_octopus_energy_2024-06-14T15-40-18.911Z.log)

### Confirmation

- [X] I confirm that I cannot find my solution within the [documentation](https://bottlecapdave.github.io/HomeAssistant-OctopusEnergy)
- [X] I confirm that I cannot find my solution within the [FAQ](https://bottlecapdave.github.io/HomeAssistant-OctopusEnergy/faq)

**Comment by paulmthomas2** [comment]
actually this is back behaving the same way again - only updating every 30 mins.

**Comment by stevenbrown1862** [comment]
I've also noticed this today.   It may have coincided with me updating home assistant.

**Comment by paulmthomas2** [comment]
> I've also noticed this today. It may have coincided with me updating home assistant.

I updated HA Core today, but only after I noticed this recurring, and it hasn't made a difference!

**Comment by Kyleidge** [comment]
Yup same issue here. Although it happened after a local powercut...

**Comment by pcb1962** [comment]
I'm seeing half-hourly updates from the Home Mini since 2pm today. 
Home Pro and Glow are reporting normally.

![image](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/5259660/de5c8f7e-3154-4f82-8f1a-d91893d3531b)

**Comment by martinpelant** [comment]
Same for me - today afternoon it started updating only once every 30 minutes - I didn't change or update anything when that happened. Since then I updated HA core and this integration but to no effect

**Comment by pantherale0** [comment]
I noticed the same issue today. No errors in logs relating to this integration.

![image](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/7346615/635bf453-70bd-46ba-9900-cadc10ffab59)

**Comment by BottlecapDave** [comment]
Hello and sorry that you're seeing this issue. I haven't personally seen this. Can you keep track of the data_last_retrieved field? This should update at the frequency that you've specified. If it is, then it's suggesting an issue with the API not updating the data. If this isnt the case, there should be warnings in your logs around using cached data. 

Another thing to bare in mind is if you're accessing the data in other ways (e.g. octowatch) as this will count towards your hourly limit which is 100 times an hour for the specific API across all meters.

**Comment by en5ads** [comment]
Hello, just to say this is happening to me too, about 14:00 yesterday and hasn't resolved yet. I don't have any other methods of accessing the data.
Anthony

**Comment by JimRnewell** [comment]
Same for me too sincde around 2pm yesterday

**Comment by Kev1507** [comment]
Same is happening for me too, also since 2pm yesterday. The data_last_retrieved attribute is updating approximately every minute though but the actual reading updates every half an hour.

**Comment by lepkirk** [comment]
Same for me, since 2pm yesterday (17/6/24) I started getting updated only every 30 mins. I haven’t made any changes to my system or anything octopus since weeks ago.

**Comment by pcb1962** [comment]
My Last Retrieved time is updating every minute as configured, I'm not running any other Octopus apps, and I've unplugged my Home Pro - no change, must be an issue at the Octopus end.

**Comment by paulmthomas2** [comment]
just to add to my comments above, i'm still getting (when i choose to look) updates every minute on the Octo-Aid app, which is driven by the API. I don't have anything else that auto-calls on the API, so it's not an API limiting matter as far as I can see.

**Comment by plord12** [comment]
Not much to add, but I'm seeing the same.  For me started at 2pm yesterday ... maybe related to when I updated to 11.2.0 ?

<img width="1549" alt="Screenshot 2024-06-18 at 11 21 21" src="https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/63543143/b91f2773-32ed-4ded-b102-96eb7b843a76">

**Comment by smeg9** [comment]
+1 for the 2PM yesterday

I updated to the latest version today, before that I was a few versions behind.

<img width="806" alt="image" src="https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/389409/a3c28482-60aa-4cb6-8915-673d2966d7d7">

**Comment by BottlecapDave** [comment]
I've raise this within the [forums](https://forum.octopus.energy/t/change-in-behaviour-for-smartmetertelemetry/10024) so awaiting a response.

**Comment by townsmcp** [comment]
Just want to add a +1 for me as well. The data_last_retrieved attribute is updating approximately every minute though but the actual reading updates every half an hour. Also only using Octo-Aid. I did have the Glow IHD working for near realtime reporting right up until saturday when I had 5 back to back power cuts over the course of an hour which seems to have blown up the IHD/smart meter comms hub :(

**Comment by smsm1** [comment]
+1 I'm also having the issue from about 2pm on 17th June 2024. With so many having the issue it does suggest a change in Octopus Energy and.

**Comment by funks81** [comment]
I’m also having this issue - have tweeted @agilephil at Octopus to see if this is an issue with Octopus.

**Comment by wralb** [comment]
I have the same symptoms starting precisely 2PM on 17th.  My Home Assistant (except the DB) is in its own VM so I rolled back to a snapshot from the 16th.  The issue was still there with this snapshot; this appears to rule out local software of config changes, and points to a change in the behaviour of the OE API.  

For info... My OE App is still providing minute by minute data.  By comparing this to hass, it seems the data being provided by OE is the **consumption for one minute** every 30 minutes (not the average over those 30 minutes); so the data is particularly misleading now.

**Comment by funks81** [comment]
So - I've had a reply from @agilephil - he doesn't suggest any problems with the Octopus API - but has suggested the issue may be the API rate limit.  I find this a little strange - as I thought the rate limit was 100 per hour - and I've only ever requested 1 minute updates - and has been working fine for the last year or so.

**Comment by paulmthomas2** [comment]
> So - I've had a reply from @agilephil - he doesn't suggest any problems with the Octopus API - but has suggested the issue may be the API rate limit. I find this a little strange - as I thought the rate limit was 100 per hour - and I've only ever requested 1 minute updates - and has been working fine for the last year or so.

It seems odd that it would be the API rate limit, bearing in mind that (should I chose too), i can watch the same minute-frequency updates in Octo-Aid (and that's still working). Note that i don't habitually do this, so I don't think it's an issue with overall API calls because i'm using them with another device/system.

**Comment by smsm1** [comment]
I wonder what API call OctoAid is using it is still working there?

**Comment by BottlecapDave** [comment]
I don't have access to the source code for octoaid, but my guess is it's doing demand and consumption separately or requesting the deltas at a granular level. Granular deltas is an option, but I'm worried about people restarting their HA instances closer to midnight that could result in up to 1440 records being returned instead of the current max of 48. To minimise the input I would need to cache the results between restarts, and be smart so I don't write too much to disk.

**Comment by funks81** [comment]
> I noticed the same issue today. No errors in logs relating to this integration.
> 
> ![image](https://private-user-images.githubusercontent.com/7346615/340467264-635bf453-70bd-46ba-9900-cadc10ffab59.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg4MTQ1ODUsIm5iZiI6MTcxODgxNDI4NSwicGF0aCI6Ii83MzQ2NjE1LzM0MDQ2NzI2NC02MzViZjQ1My03MGJkLTQ2YmEtOTkwMC1jYWRjMTBmZmFiNTkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYxOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MTlUMTYyNDQ1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MTMyMWVjYmRlMjFjNjFjYWIyMjBkZjdmMGRlZjA3NTQ0MDdkNzNjMWMzZTAxMGQ2ZDdkY2U2OGM1MDA4YzQ4ZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.CwMsfz37I2KOvRgcqllCP166Bn3f9WUPxGxhgS2wymA)

So - when I look back at my history - mine seems to have stopped working at exactly the same time as yours - just after 1400 on the 17th.  Not sure if this is relevant/exactly the same for others?

**Comment by perdimia** [comment]
This has alolao started happening on ours the last couple of days

**Comment by sopski8890** [comment]
+1 for us too. Only updating every 30 mins. Last evaluated and last retrieved updates every minute as per config, but value doesn't update

**Comment by Seaniau** [comment]
+1 Same issue, same start time. 17th June 14:00.

**Comment by sdjmchattie** [comment]
Same on ours as well:

![image](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/1999241/3a4291ec-4cce-4b1f-a1cb-bfcd59af2627)

Last value change on the minute was at 14:01:56 the 17th on our HA instance, then it switched to half hours where the value changes even though it does get pulled every minute as usual.

Other anomalies on our historic data are me resetting the network, where the line has a break and, weirdly, one instance of the value changing twice in successive minutes around 16:01.

@BottlecapDave you mentioned before that you hadn't seen this personally. Is that still the case? Just trying to understand if this is associated with our account/tariff/usage.

**Comment by Rifraferty** [comment]
> > I noticed the same issue today. No errors in logs relating to this integration.
> > ![image](https://private-user-images.githubusercontent.com/7346615/340467264-635bf453-70bd-46ba-9900-cadc10ffab59.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg4MTQ1ODUsIm5iZiI6MTcxODgxNDI4NSwicGF0aCI6Ii83MzQ2NjE1LzM0MDQ2NzI2NC02MzViZjQ1My03MGJkLTQ2YmEtOTkwMC1jYWRjMTBmZmFiNTkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYxOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MTlUMTYyNDQ1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MTMyMWVjYmRlMjFjNjFjYWIyMjBkZjdmMGRlZjA3NTQ0MDdkNzNjMWMzZTAxMGQ2ZDdkY2U2OGM1MDA4YzQ4ZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.CwMsfz37I2KOvRgcqllCP166Bn3f9WUPxGxhgS2wymA)
> 
> So - when I look back at my history - mine seems to have stopped working at exactly the same time as yours - just after 1400 on the 17th. Not sure if this is relevant/exactly the same for others?

I got the same issue which occurred just after 2pm on the 17th too

**Comment by wralb** [comment]
The environment where I'm seeing this issue:

Devices: Home Mini & Ohme Home Pro EV Charger - nothing else accessing my account I am aware of exchept very infrequent use of Octopus app and web site.
Tarrifs: E-1R-INTELLI-VAR-22-10-14-J & G-1R-LOYAL-FIX-12M-24-02-24-J
HASS: bottlecapdave integration v11.2.0 on hass 2024.6.3 in python 3.12.4 venv on ubuntu LTS 22.04.4 in VM on XCP_NG 8.3 hypervisor

Rolling back to snapshot prior to 17 June didn't help. Setting the query interval to 5 mins instead of 1 min doesn't help.

**Comment by townsmcp** [comment]
Does anyone have access to the forums at https://forum.octopus.energy/login ? Maybe others have been discussing similar regarding API's? I have just sent them an email as per the bottom of the page at https://octopus.energy/dashboard/new/accounts/personal-details/api-access

**Comment by paulmthomas2** [comment]
![image](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/157972685/06e63dbc-b4be-4891-bc96-3f5dad7ae6bc)

I've just observed that as of sometime between 17th June and end of 18th June, the cumulative consumption sensor also started only returning new information every thirty minutes (attempted to put a history graph screen shot in), which suggests this is more widespread in terms of how data is being returned from the API?

**Comment by BottlecapDave** [comment]
@sdjmchattie nope I'm seeing it as well. It feels like they changed the behaviour of how the demand is reported in a grouped scenario. I want data grouped by half hour for consumption sensors, but want the latest demand for the demand sensor. This used to work fine, but the demand now seems to be an average over the grouped period. I'd make two calls if it wasn't for the hard request limit on this API, which the integration is already pushing.

I'm still waiting for a response, but if I have time over the weekend I'll start looking at making delta calls with finer grouping (which will probably be worse for their servers in scenarios where a user is at the end of the day that needs the whole days worth of data for some reason (e.g. no internet all day)

**Comment by JuzzzyP** [comment]
I think this may now be resolved fingers crossed just started updating every 60 seconds or so. Thanks for the fine work!

**Comment by wralb** [comment]
Confirm that around 8.46 today I started getting more regular data, seems to be working as designed again. Fingers crossed.

**Comment by smsm1** [comment]
Same here that it's suddenly started working again.

**Comment by JimRnewell** [comment]
same here around 8:47am this morrning

**Comment by pcb1962** [comment]
Improved but not resolved, I'm currently seeing updates at 2 minute intervals.

**Comment by JimRnewell** [comment]
yes also every minute here

**Comment by perdimia** [comment]
Mine is also back to updating one a minute

**Comment by BottlecapDave** [comment]
Well if that's fixed, it will save me a job over the weekend :). They responded this morning saying they'll check with the relevant teams so maybe they've reverted whatever they were doing before?

**Comment by BottlecapDave** [comment]
@pcb1962 the time might not change if the value of the demand doesn't change between retrieval of data (HA thing beyond my control), as it looks like you have some updates that are 1 minute apart.

**Comment by plord12** [comment]
Yep, fixed for me as well (just as the home pro dropped through the letterbox!)

**Comment by townsmcp** [comment]
Fixed here as well

**Comment by paulmthomas2** [comment]
@BottlecapDave  - brief note of appreciation for all your hard work on this integration - do you want to leave this open for now pending whether you find anything out from Octopus with respect to the root cause, so we've got this archived for future reference, or just close it now? I'm a novice at this, so probably have the etiquette all wrong! Thanks

**Comment by BottlecapDave** [comment]
Thanks for the appreciation:) I'll leave it open for a few days just in case, but it sounded like from OE it has been resolved on their end. I might look at changing how things are done in the future, but that will be a separate ticket.

**Comment by BottlecapDave** [comment]
Closing as this seems to be fixed now
